xds/internal/balancer/cdsbalancer/cluster_handler.go

```diff
@@ -82,9 +110,12 @@ func (ch *clusterHandler) constructClusterUpdate() {
 func (ch *clusterHandler) close() {
 	ch.clusterMutex.Lock()
 	defer ch.clusterMutex.Unlock()
+	ch.rootClusterName = ""
```

Comment:
This is harmless, but since you moved it, I wonder why..  Does it need to be above the `root==nil` check?

Reply:
No. I moved it down.